span.go

```diff
@@ -113,7 +116,14 @@ func (s *spanImpl) Log(ld opentracing.LogData) {
 		ld.Timestamp = time.Now()
 	}
 
-	s.raw.Logs = append(s.raw.Logs, ld)
+	if maxLogs := s.tracer.options.MaxLogsPerSpan; maxLogs > 0 && len(s.raw.Logs) == maxLogs {
+		// We have too many logs. Overwrite the earliest log (use Logs as a circular
+		// buffer).
+		s.raw.Logs[s.numDroppedLogs%maxLogs] = ld
```

Comment:
I would like us to drop from the middle of the span log, not from the beginning. A log would read something like:
log 0
log 1
log 2
** dropped 5 events **
log 8
log 9
log 10
log 11

Since this is a significant change, we should plan to do it consistently across basictracer libraries.  I'd like to get @bhs and @yurishkuro to comment.